extensions/expression-language/Expression.cpp

```diff
@@ -183,6 +187,20 @@ Value expr_ip(const std::vector<Value>& /*args*/) {
   return {};
 }
 
+Value expr_reverseDnsLookup(const std::vector<Value>& args) {
+  std::string ip_address_str = args[0].asString();
+
+  std::chrono::steady_clock::duration timeout_duration = 5s;
+  if (args.size() > 1) {
+    timeout_duration = std::chrono::milliseconds(args[1].asUnsignedLong());
+  }
+
+  return utils::net::addressFromString(ip_address_str)
+      | utils::flatMap([timeout_duration](const auto& ip_address) { return utils::net::reverseDnsLookup(ip_address, timeout_duration);})
+      | utils::map([](const auto& hostname)-> Value { return Value(hostname); })
+      | utils::valueOrElse([&](std::error_code error_code) { throw std::runtime_error(error_code.message());});
```

Comment:
I'd throw a `std::system_error` instead. It's meant to wrap an error code, as opposed to the more general `std::runtime_error`.

Reply:
Good idea, I've changed it in https://github.com/apache/nifi-minifi-cpp/pull/1510/commits/4bb1c8fc0d9c456613ef49a2bdc33762b2caf3d5#diff-f84ab4bed4b25e41f8cb612c6444720295b6050fdb70720ada4aea04b389565aR201